vaadin-crud-flow/src/main/java/com/vaadin/flow/component/crud/Crud.java

```diff
@@ -400,6 +409,44 @@ public CrudEditorPosition getEditorPosition() {
                         CrudEditorPosition.OVERLAY);
     }
 
+    private Registration gridItemClickRegistration;
+
+    /**
+     * Sets the option to open item to edit by row click.
+     * <p>
+     * If enabled, it removes the edit column created by {@link CrudGrid}.
+     *
+     * @param clickRowToEdit {@code true} to enable it ({@code false}, by default).
+     */
+    public void setClickRowToEdit(boolean clickRowToEdit) {
+        getElement().setProperty("clickRowToEdit", clickRowToEdit);
+        Grid<E> grid = getGrid();
+
+        if (clickRowToEdit) {
+            gridItemClickRegistration = grid.addItemClickListener(e -> {
+                this.gridActiveItem = e.getItem();
+            });
+            if (hasEditColumn(grid)) {
+                removeEditColumn(grid);
+            }
+        } else if (gridItemClickRegistration != null) {
+            this.gridActiveItem = null;
+            gridItemClickRegistration.remove();
+            if (grid instanceof CrudGrid) {
+                addEditColumn(grid);
+            }
+        }
+    }
+
+    /**
+     * Gets whether click on row to edit item is enabled or not.
+     *
+     * @return the value of click on row to edit
+     */
+    public boolean getClickRowToEdit() {
```

Comment:
I guess it should be renamed to smth like `isClickRowToEdit` and docs updated as it's boolean

Reply:
Changed to `isEditOnClick`.